golang/networks/network_context.go

```diff
@@ -172,37 +161,50 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
 	}
 	resp, err := networkCtx.client.StartService(ctx, startServiceArgs)
 	if err != nil {
-		return nil, nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
+		return nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
 	}
 	logrus.Tracef("Successfully started service with Kurtosis API")
 
-	logrus.Tracef("Creating service interface...")
-	service := containerCreationConfig.GetServiceCreatingFunc()(serviceContext)
-	logrus.Tracef("Successfully created service interface")
-
-	networkCtx.services[serviceId] = service
-
-	availabilityChecker := services.NewDefaultAvailabilityChecker(serviceId, service)
-
-	return service, resp.UsedPortsHostPortBindings, availabilityChecker, nil
+	return serviceContext, resp.UsedPortsHostPortBindings,  nil
 }
 
 // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
-func (networkCtx *NetworkContext) GetService(serviceId services.ServiceID) (services.Service, error) {
-	networkCtx.mutex.Lock()
-	defer networkCtx.mutex.Unlock()
+func (networkCtx *NetworkContext) GetServiceContext(serviceId services.ServiceID, configFactory services.ContainerConfigFactory) (*services.ServiceContext, error) {
+	getServiceInfoArgs := &core_api_bindings.GetServiceInfoArgs{
+		ServiceId: string(serviceId),
+	}
+	serviceResponse, err := networkCtx.client.GetServiceInfo(context.Background(), getServiceInfoArgs)
+	if err != nil {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred when trying to get info from service '%v'",
+			serviceId)
+	}
 
-	service, found := networkCtx.services[serviceId]
-	if !found {
-		return nil, stacktrace.NewError("No service info found for ID '%v'", serviceId)
+	if serviceResponse.GetIpAddr() == "" {
+		return nil, stacktrace.Propagate(
```

Comment:
This should actually be a `stacktrace.NewError`, since we already check that `err == nil` up above. Would recommend an error text like "No error occurred when retrieving service info for service '%v', but we got an empty IP - this is VERY strange, and likely a bug in Kurtosis!"

Reply:
good catch!

Reply:
We will change this in the next PR